### DISCUSSION TITLE
Using Flowise with infinity as embeddings server

### DISCUSSION THREAD

**Discussion by highmanphil** [michaelfeil/infinity#77] [General]
Hey, I'm trying to setup https://github.com/FlowiseAI/Flowise with infinity as an embeddings server in the backend. While trying to get everything working, I got into an error namly:
Flowise is using langchain OpenAIEmbeddings and it somehow doesn't send a list of strings, when sending the prompt (it sends the input as a simple string). 
With the huggingface/text-inference it was working - possibly they implemented both request types. 
Does anyone know any kind of workaround for this kind of issue?

**Comment by michaelfeil** [comment]
I implemented a langchain version using Infinity embeddings. Its in all langchain packages since around ~0.0.330.
https://python.langchain.com/docs/integrations/text_embedding/infinity

**Reply by highmanphil** [reply]
Yeah, I already saw that - but the other project is using langchain.ts (is infinity also available in js?)

**Reply by michaelfeil** [reply]
Sadly no. Also sending list of tokens is just not a thing for non-openai models: https://github.com/michaelfeil/infinity/issues/36

**Reply by highmanphil** [reply]
So, I got it working with a small translation server in the middle
It seams like langchain in javascript just sends either input: str (if you make a query) or list[str] if you embed multiple strings. 
My server now only puts the single string into an array with the string as its only element. 
Kinda works, but isn't that beautiful
